packages/common-web/src/util.ts

```diff
@@ -17,11 +22,35 @@ export function omit(
   obj: Record<string, unknown>,
   keys: readonly string[],
 ): undefined | Record<string, unknown> {
-  if (!obj) return obj
+  // Hot path
+  if (obj) {
+    const objKeys = Object.keys(obj)
+    for (let i = 0; i < objKeys.length; i++) {
+      // If the key at position i must be excluded
+      if (keys.includes(keys[i])) {
```

Comment:
This condition looks like it's always true, so long as `keys` isn't empty—I assume we want to check `objKeys[i]` instead.

This change to `omit()` is fairly complicated, partly demonstrated by this little mistake which is easy to make and hard to catch.  Another subtle change here is that previously the caller always received a shallow clone, and now you it may not receive a shallow clone, which changes how a caller might consider handling the result.

In your estimation do you think the perf wins here are going to be measurable for us?  If it's not a win, I wonder if this is a case where we can safely reach for the simpler implementation.  If it is going to be a win, let's go for it and get some tests to document the behavior (e.g. around the shallow copying).

Reply:
I like to optimize by default but in this instance you are right, the efficiency vs complexity is not the best. Simplified accordingly.